File: src/toolchain/workload/android/platform/z42vm/src/main/java/io/z42/vm/Z42VMEntry.kt
package io.z42.vm

/**
 * Opaque handle for a resolved entry (function / static method).
 * Lifetime tied to the owning [Z42VM]; same close semantics as
 * [Z42VMModule].
 */
class Z42VMEntry internal constructor(
    @JvmField internal val nativeHandle: Long,
)

File: src/toolchain/workload/ios/platform/Sources/Z42VM/Z42VMEntry.swift
import Foundation

/// Opaque handle for a resolved entry (function / static method).
/// Lifetime tied to the owning `Z42VM`; same `.notInit` rules as
/// `Z42VMModule`.
public final class Z42VMEntry {
    let handle: OpaquePointer

    init(handle: OpaquePointer) {
        self.handle = handle
    }
}
